components/server/src/workspace/context-service.spec.db.ts

```diff
@@ -54,14 +60,111 @@ const gitpodEmptyContext = {
         private: false,
     },
     normalizedContextURL: "https://github.com/gitpod-io/empty",
-    revision: "asdf",
+    revision: "123456",
     title: "gitpod-io/empty - main",
 };
 
+// MockRepositoryProvider is a class implementing the RepositoryProvider interface, which allows to pass in commitHistory and commitInfo as needed
+
+type NamedBranch = Omit<Branch, "commit">;
+function toBranch(b: { branch: NamedBranch; commits: CommitInfo[] }): Branch {
+    return {
+        ...b.branch,
+        commit: {
+            sha: head(b.commits).sha,
+            author: head(b.commits).author,
+            commitMessage: head(b.commits).commitMessage,
+        },
+    };
+}
+class MockRepositoryProvider implements RepositoryProvider {
+    branches: Map<string, { branch: NamedBranch; commits: CommitInfo[] }> = new Map();
+
+    addBranch(branch: Omit<Branch, "commit">, commits: CommitInfo[]) {
+        this.branches.set(branch.name, {
+            branch,
+            commits,
+        });
+    }
+
+    pushCommit(branch: string, commit: CommitInfo) {
+        const b = this.branches.get(branch);
+        if (!b) {
+            throw new Error("branch not found");
+        }
+        b.commits.push(commit);
```

Comment:
For consistency with the actual implementation, let's prepend newer commits (requires other changes below this one ;)
```suggestion
        b.commits.unshift(commit);
```